src/components/product.js

```diff
@@ -65,6 +75,7 @@ export default class Product extends Component {
    * @param {Object} props - data and utilities passed down from UI instance.
    */
   constructor(config, props) {
+    config.id = normalizeProductId(config).id;
```

Comment:
This assignment is a noop. You're mutating config inside the function. You could do this without the assignment and have the same side effect.

Try to have the method receive data and return data without mutating anything so like:

```javascript
function normalizeProductId(config) {
   if (config.storefrontId) {
     return config.storefrontId;
   } else if (config.id) {
     return = btoa(`gid://shopify/Product/${config.id}`);
   } else {
    throw new Error('No ID was provided');
  }
 }

config.storefrontId = normalizeProductId(config);
```

Mutating inside the function is obfuscating the behaviour you're trying to represent. Also, since the storefront ID is the canonical ID that you're using, storefront ID should be what you're assigning. The ID returned by this is not a database id.